app/main/sauce.js

```diff
@@ -0,0 +1,57 @@
+import WebSocket from 'ws';
+
+const SAUCE_IPC_TYPES = {
+  CLOSE_WS: 'closeWebsocket',
+  RUN_WS: 'runWebsocket',
+  START_WS: 'startWebsocket',
+  WS_CLOSED: 'websocketClosed',
+  WS_ERROR: 'websocketError',
+  WS_MESSAGE: 'websocketMessage',
+  WS_STARTED: 'websocketStarted',
+};
+
+let wss = null;
+
+/**
+ * Start the websocket and process the data
+ *
+ * @param {object} event
+ * @param {object} websocketData
+ * @param {string} websocketData.accessKey
+ * @param {string} websocketData.dataCenter
+ * @param {string} websocketData.sessionId
+ * @param {string} websocketData.username
+ */
+function runWebSocket (event, { accessKey, dataCenter, sessionId, username }) {
+  const wsUrl = `wss://api.${dataCenter}.saucelabs.com/v1/rdc/socket/alternativeIo/${sessionId}`;
+  wss = new WebSocket(wsUrl, {
+    headers: {
+      Authorization: `Basic ${new Buffer.from(
+        `${username}:${accessKey}`
+      ).toString('base64')}`,
+    },
+  });
+  wss.onopen = () => {
```

Comment:
do we need opening and closing brackets for oneline hanlders?

Reply:
if we omit them, it's an implicit return, and while that may be OK in this case, I find it better to be explicit